src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ExpressionHelper.cs

```diff
@@ -178,9 +190,18 @@ public static bool IsSingleArgumentIndexer(Expression expression)
             }
 
             // Find default property (the indexer) and confirm its getter is the method in this expression.
-            return declaringType.GetRuntimeProperties().Any(
-                property => (string.Equals(defaultMember.MemberName, property.Name, StringComparison.Ordinal) &&
-                    property.GetMethod == methodExpression.Method));
+            var runtimeProperties = declaringType.GetRuntimeProperties();
+            for (int i=0; i< runtimeProperties.Count(); i++ )
```

Comment:
Again, just use a `foreach`. The `Count()` and `ElementAt()` calls iterate multiple times when this loop only needs to go through the elements once.